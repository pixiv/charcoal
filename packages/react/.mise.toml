[tools]
pnpm = "10.17.1"
