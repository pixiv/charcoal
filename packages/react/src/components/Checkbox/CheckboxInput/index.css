.charcoal-checkbox-input {
  appearance: none;
  display: flex;
  cursor: pointer;
  margin: 0;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  transition: 0.2s box-shadow, 0.2s background-color;
  position: relative;
  box-sizing: border-box;
  background-color: var(--charcoal-color-container-default);
}

.charcoal-checkbox-input:hover {
  background-color: var(--charcoal-color-container-hover);
}
.charcoal-checkbox-input:active {
  background-color: var(--charcoal-color-container-press);
}
.charcoal-checkbox-input:disabled {
  cursor: default;
}
.charcoal-checkbox-input[readonly] {
  cursor: default;
  opacity: 0.32;
}

.charcoal-checkbox-input:checked {
  background-color: var(--charcoal-color-container-primary-default);
}

.charcoal-checkbox-input:checked::after {
  content: '';
  background-color: var(--charcoal-color-icon-on-primary-default);
  display: flex;
  margin: auto;
  width: 16px;
  height: 16px;
  clip-path: path(
    'M10.6 5a1.3 1.3 0 0 1 1.8 1.9l-5.7 5.6-3-2.9a1.3 1.3 0 1 1 2-1.9l1 1z'
  );
}

.charcoal-checkbox-input:checked:hover::after {
  background-color: var(--charcoal-color-icon-on-primary-hover);
}

.charcoal-checkbox-input:checked:active::after {
  background-color: var(--charcoal-color-icon-on-primary-press);
}

.charcoal-checkbox-input:not(:checked) {
  border-width: 2px;
  border-style: solid;
  border-color: var(--charcoal-color-border-default);
}

.charcoal-checkbox-input:not(:disabled):focus {
  outline: none;
  box-shadow: var(--charcoal-color-custom-focus-effect);
}
.charcoal-checkbox-input:not(:disabled):focus-visible {
  box-shadow: var(--charcoal-color-custom-focus-effect);
}

.charcoal-checkbox-input:checked:not(:disabled, :read-only):hover {
  background-color: var(--charcoal-color-container-primary-hover);
}
.charcoal-checkbox-input:checked:not(:disabled, :read-only):active {
  background-color: var(--charcoal-color-container-primary-press);
}

.charcoal-checkbox-input:not(:disabled):focus:not(:focus-visible) {
  box-shadow: none;
}

.charcoal-checkbox-input[aria-invalid='true'],
.charcoal-checkbox-input[aria-invalid='true']:not(:disabled):focus {
  box-shadow: var(--charcoal-color-custom-error-effect);
}

.charcoal-checkbox-input[data-rounded='true'] {
  border-radius: 10px;
  background-color: var(--charcoal-color-container-secondary-default);
  border: 2px solid transparent;
}

.charcoal-checkbox-input[data-rounded='true']:checked {
  background-color: var(--charcoal-color-container-primary-default);
}

.charcoal-checkbox-input[data-rounded='true']:not(:disabled):hover {
  background-color: var(--charcoal-color-container-secondary-hover);
}
.charcoal-checkbox-input[data-rounded='true']:not(:disabled):active {
  background-color: var(--charcoal-color-container-secondary-press);
}

.charcoal-checkbox-input[data-rounded='true']:not(:disabled):focus-visible {
  box-shadow: var(--charcoal-color-custom-focus-effect);
}

.charcoal-checkbox-input[data-rounded='true']:checked:not(:disabled):hover {
  background-color: var(--charcoal-color-container-primary-hover);
}
.charcoal-checkbox-input[data-rounded='true']:checked:not(:disabled):active {
  background-color: var(--charcoal-color-container-primary-press);
}

.charcoal-checkbox-input[data-rounded='true'][aria-invalid='true'],
.charcoal-checkbox-input[data-rounded='true'][aria-invalid='true']:not(
    :disabled
  ):focus {
  box-shadow: var(--charcoal-color-custom-error-effect);
}

.charcoal-checkbox-input[data-rounded='true']::before {
  content: '';
  width: 24px;
  height: 24px;
  position: absolute;
  top: -4px;
  left: -4px;
  border-radius: 12px;
  border: 2px solid #fff;
  box-sizing: border-box;
}
