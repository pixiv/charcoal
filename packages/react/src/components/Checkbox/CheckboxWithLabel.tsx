import React from 'react'
import { useClassNames } from '../../_lib/useClassNames'

export const CheckboxWithLabel: React.NamedExoticComponent<{
  children: React.ReactNode
  input: React.ReactNode
  disabled?: boolean
  className?: string
  id?: string
}> = React.memo(function CheckboxWithLabel({
  className,
  children,
  input,
  id,
  disabled,
}: {
  children: React.ReactNode
  input: React.ReactNode
  disabled?: boolean
  className?: string
  id?: string
}) {
  const classNames = useClassNames('charcoal-checkbox__label', className)
  return (
    <label htmlFor={id} aria-disabled={disabled} className={classNames}>
      {input}
      <div className="charcoal-checkbox__label_div">{children}</div>
    </label>
  )
})
