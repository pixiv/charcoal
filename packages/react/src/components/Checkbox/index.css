.charcoal-checkbox__label {
  position: relative;
  cursor: pointer;
  display: flex;
  gap: 4px;
  color: var(--charcoal-color-text-default);
}

.charcoal-checkbox__label[aria-disabled='true'] {
  cursor: default;
  opacity: 0.32;
}

.charcoal-checkbox__label:has(input[readonly]) {
  cursor: default;
}

.charcoal-checkbox__label_div {
  color: inherit;
  font-size: 14px;
  line-height: 20px;
}
