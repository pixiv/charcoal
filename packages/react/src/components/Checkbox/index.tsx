import React, { forwardRef, memo, useMemo } from 'react'
import styled, { css } from 'styled-components'
import { useCheckbox } from '@react-aria/checkbox'
import { useObjectRef } from '@react-aria/utils'
import { useToggleState } from 'react-stately'
import { disabledSelector, px } from '@charcoal-ui/utils'
import { theme } from '../../styled'

import type { AriaCheckboxProps } from '@react-types/checkbox'
import Icon from '../Icon'

type CheckboxLabelProps =
  | {
      children: React.ReactNode
    }
  | {
      label: string
    }

export type CheckboxProps = CheckboxLabelProps & {
  readonly id?: string
  readonly name?: string

  readonly checked?: boolean
  readonly defaultChecked?: boolean
  readonly disabled?: boolean
  readonly readonly?: boolean

  readonly onClick?: () => void
  readonly onChange?: (isSelected: boolean) => void
  readonly onBlur?: () => void
  readonly onFocus?: () => void
}

const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
  function CheckboxInner(props, ref) {
    const ariaCheckboxProps = useMemo<AriaCheckboxProps>(
      () => ({
        ...props,
        isSelected: props.checked,
        defaultSelected: props.defaultChecked,
        // children がいない場合は aria-label をつけないといけない
        'aria-label': 'children' in props ? undefined : props.label,
        isDisabled: props.disabled,
      }),
      [props]
    )
    const state = useToggleState(ariaCheckboxProps)
    const objectRef = useObjectRef(ref)

    const { inputProps } = useCheckbox(ariaCheckboxProps, state, objectRef)
    const isDisabled = (props.disabled ?? false) || (props.readonly ?? false)

    return (
      <InputRoot aria-disabled={isDisabled}>
        <CheckboxInput type="checkbox" {...inputProps} />
        <CheckboxInputOverlay aria-hidden={true} checked={inputProps.checked}>
          <Icon name="24/Check" unsafeNonGuidelineScale={2 / 3} />
        </CheckboxInputOverlay>

        {'children' in props && <InputLabel>{props.children}</InputLabel>}
      </InputRoot>
    )
  }
)

export default memo(Checkbox)

const hiddenCss = css`
  visibility: hidden;
`

const InputRoot = styled.label`
  position: relative;
  display: flex;
  align-items: center;
  cursor: pointer;
  ${disabledSelector} {
    cursor: default;
  }

  gap: ${({ theme }) => px(theme.spacing[4])};
  ${theme((o) => [o.disabled])}
`

const CheckboxInput = styled.input`
  &[type='checkbox'] {
    appearance: none;
    display: block;
    cursor: pointer;
    margin: 0;
    width: 20px;
    height: 20px;

    &:checked {
      ${theme((o) => o.bg.brand.hover.press)}
    }
    &:not(:checked) {
      border-width: 2px;
      border-style: solid;
      border-color: ${({ theme }) => theme.color.text4};
    }
    ${theme((o) => [o.outline.default.focus, o.borderRadius(4)])}
  }
`
const CheckboxInputOverlay = styled.div<{ checked?: boolean }>`
  position: absolute;
  top: -2px;
  left: -2px;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;

  ${theme((o) => [o.width.px(24), o.height.px(24), o.font.text5])}

  ${({ checked }) => checked !== true && hiddenCss};
`

const InputLabel = styled.div`
  ${theme((o) => [o.typography(14)])}
`
