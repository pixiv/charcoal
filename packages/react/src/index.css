:root {
  /* charcoal custom css variables */
  --charcoal-color-custom-focus-effect: 0 0 0 4px rgba(0, 150, 250, 0.32);
  --charcoal-color-custom-error-effect: 0 0 0 4px rgba(255, 43, 0, 0.32);
}
