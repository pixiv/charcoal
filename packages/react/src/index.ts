export {
  default as ComponentAbstraction,
  useComponentAbstraction,
  type LinkProps,
} from './core/ComponentAbstraction'
export { SSRProvider } from './core/SSRProvider'
export { default as Button, type ButtonProps } from './components/Button'
export {
  default as Clickable,
  type ClickableProps,
  type ClickableElement,
} from './components/Clickable'
export {
  default as IconButton,
  type IconButtonProps,
} from './components/IconButton'
export {
  default as Radio,
  type RadioProps,
  RadioGroup,
  type RadioGroupProps,
} from './components/Radio'
export {
  default as Select,
  type SelectProps,
  SelectGroup,
  type SelectGroupProps,
} from './components/Select'
export { default as Switch, type SwitchProps } from './components/Switch'
export {
  default as TextField,
  type TextFieldProps,
  type SingleLineTextFieldProps,
  type MultiLineTextFieldProps,
} from './components/TextField'
export { default as Icon, type IconProps } from './components/Icon'
export { default as Modal, type ModalProps } from './components/Modal'
export {
  default as LoadingSpinner,
  LoadingSpinnerIcon,
} from './components/LoadingSpinner'
export {
  default as SegmentedControl,
  type SegmentedControlProps,
} from './components/SegmentedControl'
export { default as Checkbox } from './components/Checkbox'
export { default as TagItem, type TagItemProps } from "./components/TagItem"
