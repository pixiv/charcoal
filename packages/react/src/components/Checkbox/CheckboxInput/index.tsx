import './index.css'

import { forwardRef, ForwardRefExoticComponent, memo, MemoExoticComponent, RefAttributes, useCallback } from 'react'
import { useClassNames } from '../../../_lib/useClassNames'

type CharcoalCheckboxInputProps = {
  invalid?: boolean
  onChange?: (checked: boolean) => void
  rounded?: boolean
}

type InputProps = React.HTMLProps<HTMLInputElement>

export type CheckboxInputProps = CharcoalCheckboxInputProps &
  Omit<InputProps, keyof CharcoalCheckboxInputProps | 'ref'>

const CheckboxInput = forwardRef<HTMLInputElement, CheckboxInputProps>(
  function CheckboxInput(
    { onChange, checked, invalid, className, rounded, ...props },
    ref
  ) {
    const handleChange = useCallback(
      (e: React.ChangeEvent<HTMLInputElement>) => {
        const el = e.currentTarget
        onChange?.(el.checked)
      },
      [onChange]
    )

    const classNames = useClassNames('charcoal-checkbox-input', className)

    return (
      <input
        className={classNames}
        ref={ref}
        type="checkbox"
        onChange={handleChange}
        aria-invalid={invalid}
        checked={checked}
        data-rounded={rounded}
        {...props}
      />
    )
  }
)

const _default_1: MemoExoticComponent<ForwardRefExoticComponent<CharcoalCheckboxInputProps & Omit<InputProps, "ref" | keyof CharcoalCheckboxInputProps> & RefAttributes<HTMLInputElement>>> = memo(CheckboxInput)
export default _default_1
