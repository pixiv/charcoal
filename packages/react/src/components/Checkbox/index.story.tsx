import { action } from '@storybook/addon-actions'
import Checkbox from '.'
import { Meta, StoryObj } from '@storybook/react'
import { useCallback, useState } from 'react'
import styled from 'styled-components'

export default {
  title: 'Checkbox',
  component: Checkbox,
} as Meta<typeof Checkbox>

const InnerText = styled.span`
  user-select: none;
`

export const Default: StoryObj<typeof Checkbox> = {
  render: (props) => {
    const [checked, setChecked] = useState(props.checked)
    const handleChange = useCallback((isSelected: boolean) => {
      setChecked(isSelected)
      action('change')(isSelected)
    }, [])

    return (
      <Checkbox
        {...props}
        checked={checked ?? props.checked}
        name="labelled"
        label="label"
        onBlur={action('blur')}
        onClick={action('click')}
        onChange={handleChange}
        onFocus={action('focus')}
      />
    )
  },
}

export const Label: StoryObj<typeof Checkbox> = {
  render: () => {
    return (
      <Checkbox name="labelled">
        <InnerText>同意する</InnerText>
      </Checkbox>
    )
  },
}

export const Checked: StoryObj<typeof Checkbox> = {
  render: () => {
    return <Checkbox name="labelled" label="同意する" checked />
  },
}

export const Disabled: StoryObj<typeof Checkbox> = {
  render: () => {
    return <Checkbox name="labelled" label="同意する" disabled />
  },
}

export const ReadOnly: StoryObj<typeof Checkbox> = {
  render: () => {
    return <Checkbox name="labelled" label="同意する" readonly />
  },
}

export const Invalid: StoryObj<typeof Checkbox> = {
  render: () => {
    return (
      <Checkbox name="labelled" invalid>
        同意する
      </Checkbox>
    )
  },
}
