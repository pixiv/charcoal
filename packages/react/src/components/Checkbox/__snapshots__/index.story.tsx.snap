// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Checkbox Invalid 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  gap: 4px;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

.c1 {
  position: relative;
}

.c2[type='checkbox'] {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  cursor: pointer;
  margin: 0;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow,0.2s background-color;
  transition: 0.2s box-shadow,0.2s background-color;
}

.c2[type='checkbox']:checked {
  background-color: var(--charcoal-brand);
}

.c2[type='checkbox']:checked:not(:disabled):not([aria-disabled]):hover,
.c2[type='checkbox']:checked[aria-disabled='false']:hover {
  background-color: var(--charcoal-brand-hover);
}

.c2[type='checkbox']:checked:not(:disabled):not([aria-disabled]):active,
.c2[type='checkbox']:checked[aria-disabled='false']:active {
  background-color: var(--charcoal-brand-press);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus,
.c2[type='checkbox'][aria-disabled='false']:focus,
.c2[type='checkbox']:not(:disabled):not([aria-disabled]):active,
.c2[type='checkbox'][aria-disabled='false']:active {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus:not(:focus-visible),
.c2[type='checkbox'][aria-disabled='false']:focus:not(:focus-visible),
.c2[type='checkbox']:not(:disabled):not([aria-disabled]):active:not(:focus-visible),
.c2[type='checkbox'][aria-disabled='false']:active:not(:focus-visible) {
  outline: none;
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus-visible,
.c2[type='checkbox'][aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled])[aria-invalid='true'],
.c2[type='checkbox'][aria-disabled='false'][aria-invalid='true'] {
  box-shadow: 0 0 0 4px rgba(255,43,0,0.32);
}

.c2[type='checkbox']:not(:checked) {
  border-width: 2px;
  border-style: solid;
  border-color: var(--charcoal-text4);
}

.c3 {
  position: absolute;
  top: -2px;
  left: -2px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
  color: var(--charcoal-text5);
  visibility: hidden;
}

.c4 {
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 20px;
}

<div
  data-dark={false}
>
  <label
    aria-disabled={false}
    className="c0"
  >
    <div
      className="c1"
    >
      <input
        aria-invalid={true}
        checked={false}
        className="c2"
        disabled={false}
        name="labelled"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onDragStart={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        type="checkbox"
      />
      <div
        aria-hidden={true}
        checked={false}
        className="c3"
      >
        <pixiv-icon
          name="24/Check"
          unsafe-non-guideline-scale={0.6666666666666666}
        />
      </div>
    </div>
    <div
      className="c4"
    >
      同意する
    </div>
  </label>
</div>
`;

exports[`Storyshots Checkbox Labelled 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  gap: 4px;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

.c1 {
  position: relative;
}

.c2[type='checkbox'] {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  cursor: pointer;
  margin: 0;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow,0.2s background-color;
  transition: 0.2s box-shadow,0.2s background-color;
}

.c2[type='checkbox']:checked {
  background-color: var(--charcoal-brand);
}

.c2[type='checkbox']:checked:not(:disabled):not([aria-disabled]):hover,
.c2[type='checkbox']:checked[aria-disabled='false']:hover {
  background-color: var(--charcoal-brand-hover);
}

.c2[type='checkbox']:checked:not(:disabled):not([aria-disabled]):active,
.c2[type='checkbox']:checked[aria-disabled='false']:active {
  background-color: var(--charcoal-brand-press);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus,
.c2[type='checkbox'][aria-disabled='false']:focus,
.c2[type='checkbox']:not(:disabled):not([aria-disabled]):active,
.c2[type='checkbox'][aria-disabled='false']:active {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus:not(:focus-visible),
.c2[type='checkbox'][aria-disabled='false']:focus:not(:focus-visible),
.c2[type='checkbox']:not(:disabled):not([aria-disabled]):active:not(:focus-visible),
.c2[type='checkbox'][aria-disabled='false']:active:not(:focus-visible) {
  outline: none;
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus-visible,
.c2[type='checkbox'][aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled])[aria-invalid='true'],
.c2[type='checkbox'][aria-disabled='false'][aria-invalid='true'] {
  box-shadow: 0 0 0 4px rgba(255,43,0,0.32);
}

.c2[type='checkbox']:not(:checked) {
  border-width: 2px;
  border-style: solid;
  border-color: var(--charcoal-text4);
}

.c3 {
  position: absolute;
  top: -2px;
  left: -2px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
  color: var(--charcoal-text5);
  visibility: hidden;
}

.c4 {
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 20px;
}

<div
  data-dark={false}
>
  <div>
    <div
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <label
        aria-disabled={false}
        className="c0"
      >
        <div
          className="c1"
        >
          <input
            checked={false}
            className="c2"
            disabled={false}
            name="labelled"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="checkbox"
          />
          <div
            aria-hidden={true}
            checked={false}
            className="c3"
          >
            <pixiv-icon
              name="24/Check"
              unsafe-non-guideline-scale={0.6666666666666666}
            />
          </div>
        </div>
        <div
          className="c4"
        >
          同意する
        </div>
      </label>
    </div>
    <div>
      <label
        aria-disabled={false}
        className="c0"
      >
        <div
          className="c1"
        >
          <input
            checked={false}
            className="c2"
            disabled={false}
            name="labelled"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="checkbox"
          />
          <div
            aria-hidden={true}
            checked={false}
            className="c3"
          >
            <pixiv-icon
              name="24/Check"
              unsafe-non-guideline-scale={0.6666666666666666}
            />
          </div>
        </div>
        <div
          className="c4"
        >
          <span
            style={
              Object {
                "display": "block",
                "width": 200,
              }
            }
          >
            同意する同意する同意する同意する同意する同意する同意する同意する同意する同意する同意する同意する同意する
          </span>
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`Storyshots Checkbox Unlabelled 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  gap: 4px;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

.c1 {
  position: relative;
}

.c2[type='checkbox'] {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  cursor: pointer;
  margin: 0;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow,0.2s background-color;
  transition: 0.2s box-shadow,0.2s background-color;
}

.c2[type='checkbox']:checked {
  background-color: var(--charcoal-brand);
}

.c2[type='checkbox']:checked:not(:disabled):not([aria-disabled]):hover,
.c2[type='checkbox']:checked[aria-disabled='false']:hover {
  background-color: var(--charcoal-brand-hover);
}

.c2[type='checkbox']:checked:not(:disabled):not([aria-disabled]):active,
.c2[type='checkbox']:checked[aria-disabled='false']:active {
  background-color: var(--charcoal-brand-press);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus,
.c2[type='checkbox'][aria-disabled='false']:focus,
.c2[type='checkbox']:not(:disabled):not([aria-disabled]):active,
.c2[type='checkbox'][aria-disabled='false']:active {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus:not(:focus-visible),
.c2[type='checkbox'][aria-disabled='false']:focus:not(:focus-visible),
.c2[type='checkbox']:not(:disabled):not([aria-disabled]):active:not(:focus-visible),
.c2[type='checkbox'][aria-disabled='false']:active:not(:focus-visible) {
  outline: none;
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled]):focus-visible,
.c2[type='checkbox'][aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2[type='checkbox']:not(:disabled):not([aria-disabled])[aria-invalid='true'],
.c2[type='checkbox'][aria-disabled='false'][aria-invalid='true'] {
  box-shadow: 0 0 0 4px rgba(255,43,0,0.32);
}

.c2[type='checkbox']:not(:checked) {
  border-width: 2px;
  border-style: solid;
  border-color: var(--charcoal-text4);
}

.c3 {
  position: absolute;
  top: -2px;
  left: -2px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
  color: var(--charcoal-text5);
  visibility: hidden;
}

<div
  data-dark={false}
>
  <div>
    <label
      aria-disabled={false}
      className="c0"
    >
      <div
        className="c1"
      >
        <input
          aria-label="label"
          checked={false}
          className="c2"
          disabled={false}
          name="unlabelled"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          type="checkbox"
        />
        <div
          aria-hidden={true}
          checked={false}
          className="c3"
        >
          <pixiv-icon
            name="24/Check"
            unsafe-non-guideline-scale={0.6666666666666666}
          />
        </div>
      </div>
    </label>
  </div>
</div>
`;
